spaceapi_server/src/lib.rs

```diff
@@ -65,9 +65,9 @@ impl SpaceapiServer {
     /// The first argument is a ``api::SensorTemplate`` instance containing all static data.
     /// The second argument specifies how to get the actual sensor value from the datastore.
     /// And the third argument specifies the data type of the value.
-    pub fn register_sensor(&mut self, template: api::SensorTemplate, data_key: String, data_type: sensors::SensorValueType) {
+    pub fn register_sensor(&mut self, template: Box<api::SensorTemplate>, data_key: String) {
```

Comment:
To be clear a `Box<Foo>` is an *owned* trait object, one can also borrow it by using `&Foo`
(where `Foo` is a Trait obviously). See also https://doc.rust-lang.org/book/trait-objects.html#dynamic-dispatch